jung-visualization/src/main/java/edu/uci/ics/jung/visualization/spatial/Spatial.java

```diff
@@ -69,13 +68,22 @@ default Rectangle2D getUnion(Rectangle2D rect, Point2D p) {
     return new Rectangle2D.Double(left, top, right - left, bottom - top);
```

Comment:
Since you're using your own Point now, should you also be using your own Rectangle here?

Reply:
Not sure... Any points, rectangles, and other shapes on the visualization side have to be the AWT ones so that the AWT AffineTransform will work on them. Also, on the Graphics2d side, there may be better potential for GPU usage for the official types.